check/checkSpeciesTags.py

```diff
@@ -0,0 +1,106 @@
+from sys import argv
+
+re_tags = re.compile(r"^ *tags *= *\[")
+re_start_effects = re.compile(r"^ *effectsgroups *= *\[")
+re_start_foci = re.compile(r"^ *foci *= *\[")
+re_end_list = re.compile(r"^ *\]")
+re_macro = re.compile(r"^ *\[\[([A-Z_]+)\]\]")
+# Commented out are not used by the AI, most likely they never will.
+# For tags that are implemented as species effects, only the AI
+# is using the tags.
+skills_to_check = [
+    "INDUSTRY",
+    "RESEARCH",
+    "INFLUENCE",
+    # "STOCKPILE",
+    "POPULATION",
+    "HAPPINESS",
+    "SUPPLY",
+    "OFFENSE_TROOPS",
+    # "DEFENSE_TROOPS",
+    "WEAPONS",
+    "DETECTION",
+    "STEALTH",
+    "FUEL",
+]
+foci_to_check = [
+    "INDUSTRY",
+    "RESEARCH",
+    "INFLUENCE",
+]
+
+
+def check_skill(name):
+    if (
+        name.startswith("AVERAGE")
+        or name.startswith("PEDIA")
+        or name.startswith("HAEMAESTHETIC")
+        or name.startswith("INFINITE")
+    ):
+        return ""
+    for skill in skills_to_check:
+        if name.endswith(skill):
+            return skill
+    return ""
+
+
+def check_species(file):
+    effects = False
+    foci = False
+    tags = set()
+    foci_list = set(foci_to_check)
+    with open(file, "r", encoding="utf-8") as f:
+        for line in f.readlines():
+            if re_tags.match(line):
+                strings = line.split('"')
+                for i in range(1, len(strings), 2):
+                    if check_skill(strings[i]):
+                        tags.add(strings[i])
+                alltags = copy(tags)
+            if re_start_effects.match(line):
+                effects = True
+            if re_start_foci.match(line):
+                foci = True
+            if effects or foci:
+                r = re_macro.match(line)
+                if r:
+                    value = r.groups()[0]
+                    if effects:
+                        skill = check_skill(value)
+                        if skill:
+                            if value in tags:
+                                tags.remove(value)
+                            else:
+                                print("%s: %s in effectgroups, missing in tags" % (file.name, value))
+                    else:
+                        focus = value.split("_")[1]
+                        if focus in foci_list:
+                            foci_list.remove(focus)
+                            if ("NO_" + focus) in alltags:
+                                print("%s: %s, but not tag NO_%s" % (file.name, value, focus))
+                elif re_end_list.match(line):
+                    effects = False
+                    foci = False
+    for focus in foci_list:
+        if ("NO_" + focus) not in alltags:
+            print("%s: Not HAS_%s_FOCUS, NO_%s missing in tags" % (file.name, focus, focus))
+    if tags:
+        for tag in tags:
+            print("%s: %s in tags, missing in effectgroups" % (file.name, tag))
+
+
+if __name__ == "__main__":
+    if len(argv) != 2:
```

Comment:
I'd recommend using https://docs.python.org/3/library/argparse.html.

Reply:
As long as I only need exactly one value, having no options, this seemed to be the simplest solution.

Reply:
Maybe a few more lines of code, but it will greatly improve readability. Still ok to leave it as is.